src/cpp/security/accesscontrol/DistinguishedName.cpp

```diff
@@ -0,0 +1,389 @@
+    // NOTE: using -1 because size_t cannot be lower than 0, so comparison i >= 0 make no sense
+    while (i > 0 && value[i - 1] == ' ')
+    {
+        i--;
+    }
+    return cut(0, i);
+}
+
+size_t Attribute::clear() noexcept
+{
+    value = nullptr;
+    size_t aux = length;
+    length = 0;
+    return aux;
+}
+
+size_t Attribute::find(
+        char c) const noexcept
+{
+    size_t i = 0;
+    bool in_quotes = false;
+    while (i < length)
+    {
+        if (!in_quotes && value[i] == c)
+        {
+            return i;
+        }
+
+        if (value[i] == '"')
+        {
+            in_quotes = !in_quotes;
+        }
+
+        else if (value[i] == '\\')
+        {
+            i++;
+        }
+
+        i++;
+    }
+    return i;
+}
+
+size_t Attribute::find(
+        char c,
+        bool& result) const noexcept
+{
+    auto res = find(c);
+    result = found(res);
+    return res;
+}
+
+bool Attribute::found(
+        const size_t& find_result) const noexcept
+{
+    return find_result < length;
+}
+
+bool Attribute::operator ==(
+        const Attribute& other) const noexcept
+{
+    // Compare char to char
+    // scaped chars with \ must be taken into account, and " must be ignored
+    size_t index_this = 0;
+    size_t index_other = 0;
+    while (true)
+    {
+        while (
+            index_this < this->length
+            && (this->value[index_this] == '\\' || this->value[index_this] == '"'))
+        {
+            index_this++;
+        }
+
+        while (
+            index_other < other.length
+            && (other.value[index_other] == '\\' || other.value[index_other] == '"'))
+        {
+            index_other++;
+        }
+
+        if (index_this == this->length && index_other == other.length)
+        {
+            return true;
+        }
+
+        if (
+            !(index_this < this->length)
+            || !(index_other < other.length)
+            || (this->value[index_this] != other.value[index_other]))
+        {
+            return false;
+        }
+        else
+        {
+            index_this++;
+            index_other++;
+        }
+    }
+}
+
+bool Attribute::operator !=(
+        const Attribute& other) const noexcept
+{
+    return !operator ==(other);
+}
+
+Attribute Attribute::cut(
+        const Attribute& att,
+        size_t ini,
+        size_t fin /*= 0*/) noexcept
+{
+    Attribute res {att};
+    res.cut(ini, fin);
+    return res;
+}
+
+bool DistinguishedNameSpecialized::compare(
+        const DistinguishedName& name1,
+        const DistinguishedName& name2) noexcept
+{
+    DistinguishedNameSpecialized dns1(name1);
+    DistinguishedNameSpecialized dns2(name2);
+    return DistinguishedNameSpecialized(name1) == DistinguishedNameSpecialized(name2);
+}
+
+DistinguishedNameSpecialized::DistinguishedNameSpecialized(
+        const DistinguishedName& name) noexcept
+{
+    // Create it by recursive function
+    auto res = find_and_add_type_values(name);
+
+    switch (res)
```

Comment:
NIT 
Explicitly define the `ok` case would be nice. Right now it just breaks, but could be good defining it if something more needs to be added in the case

Reply:
I do not agree. If nothing should be done (or nothing different than default) the case should not be added. If at any point is required, it would be added then, without affecting the API or ABI of the function.